test/src/blockMonitor/component.test.ts

```diff
@@ -102,3 +157,21 @@ describe("MappedStateReducer", () => {
         expect(reducedState.items).to.deep.equal(expectedMap);
     });
 });
+
+describe("BlockNumberReducer", () => {
+    it("getInitialState sets current block number", () => {
+        const reducer = new BlockNumberReducer();
+        const anchorState = reducer.getInitialState(blocks[0]);
+
+        expect(anchorState.blockNumber).to.equal(blocks[0].number);
+    });
+
+    it("reduce sets current block number", () => {
+        const reducer = new BlockNumberReducer();
+
+        const prevAnchorState = reducer.getInitialState(blocks[0]);
+        const nextAnchorState = reducer.reduce(prevAnchorState, blocks[2]);
```

Comment:
It doesn't matter much here, but I think it's better to explicitly set `prevAnchorState = { ... }` instead of using `getInitialState`.